jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -72,313 +77,287 @@ public String getAttributeValue()
      * Create new HttpCookie from specific values and attributes.
      *
      * @param name the name of the cookie
      * @param value the value of the cookie
-     * @param domain the {@code Domain} value used for Domain-Matching rules on the cookie
-     * @param path the {@code Path} value to use for Path-Matching rules on the cookie
-     * @param maxAge the {@code Max-Age} attribute value (in seconds) for the cookie
-     * @param httpOnly the {@code HttpOnly} attribute of the cookie
-     * @param secure the {@code Secure} attribute of the cookie
-     * @param comment the comment of the cookie (not used in RFC6265 or Servlet 6+, only in RFC2965 mode)
-     * @param version the version of the cookie (not used in RFC6265 or Servlet 6+, only in RFC2965 mode)
-     * @param attributes the map of attributes to use with this cookie (this map is copied over into the resulting HttpCookie, but without map entries that are also parameters on this constructor)
+     * @param attributes the map of attributes to use with this cookie (this map is used for field values
+     *   such as {@link #getDomain()}, {@link #getPath()}, {@link #getMaxAge()}, {@link #isHttpOnly()},
+     *   {@link #isSecure()}, {@link #getComment()}.  These attributes are removed from the stored
+     *   attributes returned from {@link #getAttributes()}.
      */
-    public HttpCookie(String name, String value, String domain, String path, long maxAge, boolean httpOnly, boolean secure, String comment, int version, Map<String, String> attributes)
+    static HttpCookie from(String name, String value, Map<String, String> attributes)
     {
-        _name = name;
-        _value = value;
-        _domain = domain;
-        _path = path;
-        _maxAge = maxAge;
-        _httpOnly = httpOnly;
-        _secure = secure;
-        _comment = comment;
-        _version = version;
-        _expiration = maxAge < 0 ? -1 : NanoTime.now() + TimeUnit.SECONDS.toNanos(maxAge);
-        Map<String, String> attrs = null;
-        if (attributes == null || attributes.isEmpty())
-            attrs = Collections.emptyMap(); // unmodifiable empty map
-        else
-        {
-            // create new map, to only capture relevant attributes
-            attrs = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
-            attrs.putAll(attributes);
-            PARSED_ATTRIBUTE_NAMES.forEach(attrs::remove); // remove names that are also fields
-            attrs = Collections.unmodifiableMap(attrs); // don't allow attributes to be modified
-        }
-        _attributes = attrs;
+        return new Immutable(name, value, 0, attributes);
     }
 
     /**
      * Create new HttpCookie from specific values and attributes.
      *
      * @param name the name of the cookie
      * @param value the value of the cookie
-     * @param version the version of the cookie (not used in RFC6265 or Servlet 6+, only in RFC2965 mode)
+     * @param version the version of the cookie (only used in RFC2965 mode)
```

Comment:
This javadoc change worries me.
Jetty, running on Servlet 6, can still be configured for RFC2965, but this field (version) won't make it through the Servlet API down to Jetty.
That's why I worded the original Javadoc the way I did.

Reply:
This is core, so we shouldn't really codify any specific servlet version here.  And in fact we don't. We only look at the version in RFC2965 mode.  If Servlet 6 doesn't allow a non zero version to be set, then that is its business... it does not change the fact that we only look at it in one mode.     I felt it was clearer just to say what this class did with it.